packages/amazon-ssm-agent/amazon-ssm-agent.spec

```diff
@@ -17,37 +12,73 @@ Source0: %{gorepo}-%{version}.tar.gz
 Source1000: clarify.toml
 
 BuildRequires: %{_cross_os}glibc-devel
+Requires: %{name}(binaries)
 
 %description
 %{summary}.
 
+%package bin
+Summary: Remote management agent binaries
+Provides: %{name}(binaries)
+Requires: (%{_cross_os}image-feature(no-fips) and %{name})
+Conflicts: (%{_cross_os}image-feature(fips) or %{name}-fips-bin)
```

Comment:
Each binary package provides the `%{name}(binaries)` capability that the top-level package needs.

It requires the top-level package (so that `dnf install bottlerocket-runc-bin` also installs `bottlerocket-runc`) as well as the corresponding image feature capability from the metadata package. This will either be "fips" or "no-fips" depending on the variant.

Finally, it conflicts with the opposite image feature or the other binary package. The conflict helps the package manager eliminate the "wrong" package from consideration, and also prevents `bottlerocket-runc-bin` and `bottlerocket-runc-fips-bin` from being installed together.